### PR TITLE
Fixes: FallBack to LoadDataset, Dup Arguments, Savetensor Saving

### PR DESCRIPTION
**Pull request by fabianlim**
Companion PR to #6, and some additional fixes

- `datasets.load_dataset` does a schema check and sometimes may fail on some files. This fallsback to pandas 
- remove duplicated arguments
- make `savetensor` saving as default

~**API CHANGE**:~
~if the dataset has `tools` or `documents` columns in the dataset, add this to the arguments~

### CODE REVIEW

**Comment by fabianlim** [comment]
@garrett361 `TestPaddingFree` is failing because I think the runners do not have `cuda`

```python
def test_padding_free(self, model_name: str) -> None:
        torch.manual_seed(42)
>       model, cfg = self.get_fa2_model_and_cfg(model_name)
```

**Comment by garrett361** [comment]
Yeah expected, I should add a skip if CUDA isn't available

**Comment by fabianlim** [comment]
@garrett361 it seems they now run all tests.. previously they only run one test

Previously it was like this https://github.com/garrett361/open-instruct/pull/25
```
Run pytest open_instruct/test_utils.py
============================= test session starts ==============================
platform linux -- Python 3.10.10, pytest-8.4.1, pluggy-1.6.0
rootdir: /home/runner/work/open-instruct/open-instruct
configfile: pyproject.toml
collected [7](https://github.com/garrett361/open-instruct/actions/runs/16009216897/job/45162936130#step:5:8) items

open_instruct/test_utils.py .......                                      [100%]

============================== 7 passed in 14.[8](https://github.com/garrett361/open-instruct/actions/runs/16009216897/job/45162936130#step:5:9)5s ==============================
```

**Comment by garrett361** [comment]
Thought I saw that as well

**Comment by Swanand-Kadhe** [comment]
It is not necessary to include `tools` `documents` in `dataset_target_columns`

If we look at the transformation [function](https://github.com/garrett361/open-instruct/blob/5fc2ec317df6d0776c7726b19f5a07411685e5ff/open_instruct/dataset_transformation.py#L1252), first tokenization and masking happens, and then columns are removed. `tools` and `documents` columns are necessary for correctly applying the chat template (and thereby for tokenization and masking). However, after applying chat template and tokenization, the only columns needed are `input_ids`, `labels`, and `attention_mask`.

Of course, including `tools` and `documents` in `dataset_target_columns` does not impact the correctness, it will just increase the size of cached data. 

Also, `tools` and `documents` are going to be ignored in default o-i `sft_tulu_tokenize_and_truncate_v1`. In porting changes to main, we do not have ported `additional_inputs` in `sft_tulu_tokenize_and_truncate_v1`.

**Comment by fabianlim** [comment]
@garrett361 can we merge this soon? we just ignore the failing test?

File: open_instruct/dataset_transformation.py [hunk 1]
```diff
@@ -67,6 +67,38 @@
 
 from open_instruct.utils import hf_whoami
 
+# MOVE THIS SOMEWHERE
+from datasets import load_dataset as lds
```
Comment (garrett361) [inline open_instruct/dataset_transformation.py:71]: nit: move the `lds` import inside `load_dataset` so it's not exposed?

File: open_instruct/finetune.py [hunk 2]
```diff
@@ -341,13 +338,7 @@ class FlatArguments:
             "help": "Whether to clean up all previous checkpoints at the end of the run.",
         },
     )
-    final_lr_ratio: Optional[float] = field(
```
Comment (garrett361) [inline open_instruct/finetune.py:344]: Same question here, why is `final_lr_ratio` removed?

File: open_instruct/model_utils.py [hunk 3]
```diff
@@ -394,6 +394,7 @@ def save_with_accelerate(
     output_dir: str,
     use_lora: bool = False,
     model_attribute_to_save: Optional[str] = None,
+    safe_serialization: bool = True,
```
Comment (garrett361) [inline open_instruct/model_utils.py:397]: nit: make this configurable in FlatArguments?

File: open_instruct/dataset_transformation.py [hunk 4]
```diff
@@ -67,6 +67,38 @@
...
+        reader = (
+            partial(pd.read_json , lines=True)
+            if file_type in {'json', 'jsonl'}
+            else partial(pd.read_parquet, engine='auto')
+        )
+        for file in train_files:
+            dfs.append(reader(file, orient='records'))
+
+        raw_datasets = Dataset.from_pandas(pd.concat(dfs))
+        del dfs
```
Comment (garrett361) [inline open_instruct/dataset_transformation.py:100]: nit: don't need to `del dfs` since it goes out of scope immediately, right?

File: open_instruct/model_utils.py (hunk 3 above)
Comment (fabianlim) [reply open_instruct/model_utils.py:397]: if we do this then should we do it in another PR?

Comment (garrett361) [reply open_instruct/model_utils.py:397]: let's just ignore it for now, actually
